build.sh

```diff
@@ -217,7 +217,7 @@ build_native()
         pushd "$intermediatesForBuild"
         # Regenerate the CMake solution
         echo "Invoking \"$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh\" \"$__ProjectRoot\" $__ClangMajorVersion $__ClangMinorVersion $platformArch $__BuildType $__CodeCoverage $__IncludeTests $generator $extraCmakeArguments $__cmakeargs"
-        "$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh" "$__ProjectRoot" $__ClangMajorVersion $__ClangMinorVersion $platformArch $__BuildType $__CodeCoverage $__IncludeTests $generator "$extraCmakeArguments" "$__cmakeargs"
+        "$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh" "$__ProjectRoot" $__ClangMajorVersion $__ClangMinorVersion $platformArch $__BuildType $__CodeCoverage $__IncludeTests $generator "$extraCmakeArguments $__cmakeargs"
```

Comment:
I think the underlying issue might be 

https://github.com/dotnet/coreclr/blob/master/src/pal/tools/gen-buildsys-clang.sh#L45
and/or
https://github.com/dotnet/coreclr/blob/master/src/pal/tools/gen-buildsys-clang.sh#L166

Basically the extra ninja argument shifted the arguments and it broke things in this script.

I am really not in a position to test either of these as I do not have a platform which runs ninja anymore.  I am happy to close this PR and/or open an Issue.  Please advise.